### DISCUSSION TITLE
Is there any possibility to have Custom external HTTP/HTTPS ports Virtual host based ?

### DISCUSSION THREAD

**Discussion by Akshay-Hegde** [nginx-proxy/nginx-proxy#2192] [Ideas]
As of now we can change globally Custom external HTTP/HTTPS ports.

Is there any way to have Custom external HTTP/HTTPS ports for each Virtual Host if external port is different from the global ?

The existing feature path based routing is useful but difficult to enforce ACL.

Is there any way to generate something like below

```
server {
    server_name a.example.com;
    access_log /var/log/nginx/access.log vhost;
    listen 443 ssl http2 ;
   proxy_pass http://container-1:<virtual_port>/;
}
server {
    server_name a.example.com;
    access_log /var/log/nginx/access.log vhost;
    listen 8443 ssl http2 ;
   proxy_pass http://container-2:<virtual_port>/;
}
```

**Comment by rhansen** [comment]
Not yet.  See #1504.
